R/assertions.R

```diff
@@ -162,18 +165,34 @@ validate_cols <- function(data,
 
   assertr_function <- get_assert_method(predicate)
 
-  assertr_function(
-    data,
-    predicate,
-    ...,
-    skip_chain_opts = skip_chain_opts,
-    obligatory = obligatory,
-    description = description,
-    success_fun = success_fun,
-    error_fun = error_fun,
-    defect_fun = defect_fun
-  ) %>%
-    check_assertr_expression(data, description, error_fun)
+  assertion_command <- if (is.null(match.call(expand.dots = FALSE)$`...`)) {
```

Comment:
I don't like this part of code. I see that it can be improved on 3 levels, sorted with the increasing complexity of change:
1. How about moving `if` to the next line? It is hard to read right now.
2. We should extract the check to the separate variable with a meaningful name of what we are checking. For example:
```
is_colums_empty <- is.null(match.call(expand.dots = FALSE)$`...`)
assertion_command <- if(is_colums_empty) {
...
} else {
...
}
``` 
3. Maybe we could just have the columns selection in `if` clause, and have single call of the whole `assertr_function`? It might not be trivial to pass arguments correctly, so would require a bit work, up to you whether worth it.

Reply:
1. As we talked about, with nr 2 in place, this is not necessary.
2. Sounds ok
3. I was thinking about this, but ultimately, I decided to use the large if statement.  An attempt to pass just  `...`/`everything()` would overcomplicate the code.